pkg/ddevapp/ddevapp.go

```diff
@@ -1205,9 +1205,23 @@ func GetActiveApp(siteName string) (*DdevApp, error) {
 	// Mostly ignore app.Init() error, since app.Init() fails if no directory found. Some errors should be handled though.
 	// We already were successful with *finding* the app, and if we get an
 	// incomplete one we have to add to it.
-	err = app.Init(activeAppRoot)
-	if err != nil && (strings.Contains(err.Error(), "is not a valid hostname") || strings.Contains(err.Error(), "is not a valid apptype") || strings.Contains(err.Error(), "config.yaml exists but cannot be read.") || strings.Contains(err.Error(), "a project (web container) in ")) {
-		return app, err
+	if err = app.Init(activeAppRoot); err != nil {
```

Comment:
This is a massive improvement, great with the error typing. 

It will read a lot better with `case webContainerExists, invalidHostname, invalidAppType, ... : return app, err`

This is absolutely suspect code though, quite fragile, and as you'd expect it has a lot of history. I'd *rather* we had the types of error we're going to *ignore* here instead of the types we're going to return. That would be less fragile. We could do that either this time or next time.